rocks/RocksStorage.java

```diff
@@ -181,9 +181,11 @@ public Cache(OptimisticTransactionDB rocksDB) {
 
         @Override
         public <G> ResourceIterator<G> iterate(byte[] key, BiFunction<byte[], byte[], G> constructor) {
-            assert isOpen();
```

Comment:
one thing here @alexjpwalker  is that we should ask @haikalpribadi  why this was an assert not a proper exceptino before :) there might be solid reasoning it shouldn't be possible to ever reach this case (barring any new race conditions we uncovered!)

Reply:
I will provide background here for @haikalpribadi :

The bug fixed by this PR was reproduced by running, from a client, a `match; delete; insert` query, and closing the session before the query had a chance to run to completion.

On investigating the segfault stack trace me and Joshua discovered it was in the schema graph. He noticed that the "is storage closed" checks (that he added for data graph operations in https://github.com/graknlabs/grakn/pull/6185) were not present in the schema operations, so we added them for schema.